### PR TITLE
Calls with bad data should return an error response

### LINKED ISSUE #385: Calls with bad data to JSON RPC Relay should return an error response
### Description

I made a `eth_sendRawTransaction` call with bad data but did not receive an error response (receives 200 success). This may be expected behavior (see repro below).

### Steps to reproduce

1. Make a call to `eth_sendRawTransaction` with data similar to below - updated according to the nonce of the sending account and signed:

```
{
  data: '0x06e8fac2000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000003666f6f0000000000000000000000000000000000000000000000000000000000',
  to: '0x000000000000000000000000000000000214db10',
  from: '0x205399A519D341051E86e790AaA0F713340745CA',
  type: 2,
  maxFeePerGas: BigNumber { _hex: '0x026705ab4f00', _isBigNumber: true },
  maxPriorityFeePerGas: BigNumber { _hex: '0x59682f00', _isBigNumber: true },
  nonce: 87,
  gasLimit: BigNumber { value: "30678687678687676876786786876876876000" },
  chainId: 296
}
```

2. Response is not an error, instead it is:

```
{
  reason: 'overflow',
  code: 'NUMERIC_FAULT',
  fault: 'overflow',
  operation: 'toNumber',
  value: '30678687678687676876786786876876876000'
}
```

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.3.0

### Operating system

macOS

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

**Related issue(s)**:

Fixes #385

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ar-conmit — DISMISSED** [review]
LGTM

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -657,9 +657,12 @@ export class EthImpl implements Eth {
...
       await this.precheck.sendRawTransactionCheck(transaction, gasPrice);
-    }
-    catch(e: any) {
-      return e;
+    } catch (e: any) {
+      if (e instanceof JsonRpcError) {
+        return e;
+      }
+
+      throw predefined.INTERNAL_ERROR;
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:665]: There should be an `eth.spect.ts` test to highlight the issue the bug highlighted

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Missing test.

**Review by Ivo-Yankov — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**